application/src/main/java/org/togetherjava/tjbot/features/basic/MemberCounter.java

```diff
@@ -0,0 +1,48 @@
+package org.togetherjava.tjbot.features.basic;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.channel.concrete.Category;
+
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.features.Routine;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static java.lang.Integer.parseInt;
+
+public class MemberCounter implements Routine {
+    private final String wordsInCategory;
+
+    private final String updateMemberCountEveryHour;
+
+    public MemberCounter(Config config) {
+        this.wordsInCategory = config.getMemberCountCategoryName();
+        this.updateMemberCountEveryHour = config.getupdateMemberCountEveryHour();
+    }
+
+    private void updateCategoryName(Category category) {
+        int totalMemberCount = category.getGuild().getMemberCount();
+        String basename = category.getName();
+        category.getManager()
+            .setName("%s - %d Members".formatted(basename, totalMemberCount))
+            .queue();
+    }
+
+    @Override
+    public Schedule createSchedule() {
+        return new Schedule(ScheduleMode.FIXED_RATE, 0, parseInt(updateMemberCountEveryHour),
+                TimeUnit.HOURS);
+    }
+
+    @Override
+    public void runRoutine(JDA jda) {
+        jda.getGuilds()
+            .stream()
+            .map(Guild::getCategories)
+            .flatMap(List::stream)
+            .filter(cat -> cat.getName().equalsIgnoreCase(wordsInCategory))
```

Comment:
the way ur queue works is a problem. u flatten all guild channels and run it on all matching channels.

what u want instead is to run it on a single (findAny) matching channel per guild. so u need to get rid of the flattening and then use findAny per guild

Reply:
Didn't quite get what you meant here can you please elaborate thank you

Reply:
Something like this ? i used .findFirst() instead of findany
```java
@Override
    public void runRoutine(JDA jda) {
        jda.getGuilds().stream().forEach(guild -> {
            Optional<Category> categoryOptional = guild.getCategories()
                .stream()
                .filter(cat -> wordsInCategory.test(cat.getName()))
                .findFirst();
            categoryOptional.ifPresent(this::updateCategoryName);
        });
    }
```

Reply:
yeah. you can keep it a single flow btw, no need for that variable. `...findFirst().ifPresent(...)`